client.go

```diff
@@ -56,10 +57,73 @@ type APIDetails struct {
 	Details string `json:"details,omitempty"`
 }
 
-type errorObject struct {
-	Code    int         `json:"code,omitempty"`
-	Message string      `json:"message,omitempty"`
-	Errors  interface{} `json:"errors,omitempty"`
```

Comment:
Sorry for my ignorance, but figured I would ask, just to make sure. Wouldn't removing `errorObject` be a breaking change?

Reply:
@stmcallister only capitalized identifiers are exported (visible) outside of the package, and so `errorObject` could only ever be used by the internals of this project. That should make it totally safe for us to change.

In the interest of complete context sharing... we are _effectively_ renaming this to `APIErrorObject`, and so once we merge / release this any removals or breaking changes to that type would no longer be acceptable because of it being capitalized (exported).

Reply:
Cool. Thanks for the explanation! This looks great. Going to approve.